src/core/lib/slice/slice_internal.h

```diff
@@ -327,4 +327,11 @@ inline grpc_slice grpc_slice_from_static_string_internal(const char* s) {
   return grpc_slice_from_static_buffer_internal(s, strlen(s));
 }
 
+// grpc_slice_buffer_tiny_add attempts to piggy back onto an existing inlined
+// tail slice; to do this it necessarily needs to check if the slice buffer is
+// empty first. However, hpack_encoder (the main user for that method) is always
+// called with a non-empty buffer, so we can save a comparison.
+uint8_t* grpc_slice_buffer_tiny_add_non_empty_buf(grpc_slice_buffer* sb,
```

Comment:
As we discussed, my main concern here is that we need to stop adding specialized variants of every method, because it's making it impossible to know which one to use in any given situation.  I think we instead need to move in a direction where there is just one way to use each API, and the API automatically does the right thing with each type of input.

I'm fine relying on Yash and Vijay to review the HTTP/2 code, so if we can revert this API change without breaking the HTTP/2 changes, then I'm fine with this PR.

Reply:
Sure thing; I'll remove the variant and wait for @yashykt for final sign off.

Reply:
Removed.